File: benchmarks/correctness/r/csv_groupby.R
# Group CSV by cohort, compute count and mean age. Output JSON.
library(jsonlite)

meta <- read.csv("data/metadata.csv")
groups <- split(meta, meta$cohort)
results <- list()
for (cohort in sort(names(groups))) {
    g <- groups[[cohort]]
    results[[cohort]] <- list(
        count = nrow(g),
        mean_age = round(mean(g$age), 6)
    )
}

cat(toJSON(list(groups = results), auto_unbox = TRUE, pretty = TRUE, digits = 10))
cat("\n")

File: benchmarks/correctness/python/csv_groupby.py
"""Group CSV by cohort, compute count and mean age. Output JSON."""
import json
import sys
import csv
from collections import defaultdict

groups = defaultdict(list)
with open("data/metadata.csv") as f:
    reader = csv.DictReader(f)
    for row in reader:
        groups[row["cohort"]].append(float(row["age"]))

results = {}
for cohort in sorted(groups.keys()):
    ages = groups[cohort]
    results[cohort] = {
        "count": len(ages),
        "mean_age": round(sum(ages) / len(ages), 6)
    }

json.dump({"groups": results}, sys.stdout, indent=2)
print()
